### PR TITLE
DDF-04842 Migrate menu.tsx to styled and parameterize dropdown.js anchor component

### PR DESCRIPTION
**Pull request by adimka**
#### What does this PR do?
Migrate menu.tsx to styled and parameterize dropdown.js anchor component
#### Who is reviewing it? 

@zta6 @middlets719 @samuelechu 
#### Select relevant component teams: 
@codice/ui 
#### Ask 2 committers to review/merge the PR and tag them here.

@bdeining 
@djblue

#### How should this be tested?
Open up the query menu - select advanced query and select anyGeo.
The Location Dropdown uses this component. Ensure it renders normally, and that changing the value between location types is handled correctly.

#### What are the relevant tickets?
For Jira:
[](https://codice.atlassian.net/browse/)

For GH Issues:
Fixes: #4842

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by djblue** [comment]
build now

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/dropdown/dropdown.js [hunk 1]
```diff
@@ -206,4 +203,20 @@ class Dropdown extends React.Component {
   }
 }
 
+const Dropdown = ({ label, anchor, ...props }) => {
+  const dropdownProps = {
```
Comment (samuelechu) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/dropdown/dropdown.js]: Is this necessary? You could conditionally render within DropdownBody as well.

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/menu/menu.tsx [hunk 2]
```diff
@@ -1,10 +1,53 @@
...
+    position: absolute;
+    top: 50%;
+    right: 0px;
+    width: 2.275rem;
+    text-align: center;
+    transform: translateY(-50%);
+  }
+`
+
+const ItemRoot = styled.div<any>`
```
Comment (samuelechu) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/menu/menu.tsx]: what does adding `<any>` do?

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/menu/menu.tsx [hunk 3]
```diff
@@ -1,10 +1,53 @@
...
+  user-select: none; /* Non-prefixed version, currently supported by Chrome and Opera */
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  ${({theme, active}) => active ? `box-shadow: inset 0px 0px 0px 1px  ${theme.primaryColor};`: ''}
+  ${({selected}) => selected ? 'font-weight: bold;': ''}
+  ${({selected}) => selected ? after : ''}
+`
+
 class DocumentListener extends React.Component<any, any> {
```
Comment (samuelechu) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/menu/menu.tsx]: `<any, any>` 🤔 Would it be worth adding some prop / state types to Menu and MenuItem?

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/menu/menu.tsx [hunk 4]
```diff
@@ -112,10 +155,10 @@ export class Menu extends React.Component<any, any> {
     )
 
     return (
-      <Component>
+      <MenuRoot className={this.props.className}>
```
Comment (samuelechu) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/menu/menu.tsx]: What do you need className for?

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/dropdown/dropdown.js (hunk 1 above)
Comment (adimka) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/dropdown/dropdown.js]: I think this is fine either way - adding it outside the Dropdown body should help clarify that this isn't actually part of the body, and is just the part of the dropdown that you click to open it.

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/menu/menu.tsx (hunk 2 above)
Comment (adimka) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/menu/menu.tsx]: Gives access to non-theme props. I can add a type here.

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/menu/menu.tsx (hunk 4 above)
Comment (adimka) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/menu/menu.tsx]: Other components using this are still relying on less - this allows us to use styled without breaking others.

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/menu/menu.tsx (hunk 3 above)
Comment (adimka) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/menu/menu.tsx]: I'm going to leave it as is for now - we'd have to go around and make sure the changes work for everyone.

**Comment by bdeining** [comment]
build now

**Comment by andrewzimmer** [comment]
@adimka looks like this needs to be reformatted

**Comment by adimka** [comment]
build now

**Comment by bdeining** [comment]
hero successful

**Comment by bdeining** [comment]
build now

**Pull request merged**